java-engine/src/main/java/io/getunleash/engine/CustomStrategies.java

```diff
@@ -0,0 +1,151 @@
+package io.getunleash.engine;
+
+import static java.util.function.Function.identity;
+import static java.util.stream.Collectors.toMap;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.util.*;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+class CustomStrategies {
+    private static final Logger log = LoggerFactory.getLogger(CustomStrategies.class);
+    static final String EMPTY_STRATEGY_RESULTS = "{}";
+    private final Map<String, IStrategy> registeredStrategies;
+    private final ObjectMapper mapper;
+
+    private Map<String, List<MappedStrategy>> featureStrategies = new HashMap<>();
+
+    public CustomStrategies(Stream<IStrategy> customStrategies) {
+        this.mapper = new ObjectMapper();
+        this.mapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
+        this.registeredStrategies = customStrategies.collect(toMap(IStrategy::getName, identity()));
+    }
+
+    public void takeState(String toggles) {
```

Comment:
Also think this name could use some love. updateCustomStrategies maybe? It's too close to the Yggdrasil core takeState to my eyes

Reply:
Yes, it was intentional, because it happens on take state. I don't like that CustomStrategies now have to know about the shape of toggles input, because that's duplicated in Yggdrasil and will be replicated in all Engines... I was thinking how to solve this problem but we should probably talk about this

Reply:
Yeah I hate it too. I don't have a better suggestion right now though. My ideal scenario is we throw away custom strategies which will make this go away

This is basically a hack to get us over the line and put us in a place where we can do that